build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/buildmodel/BuildDescription.java

```diff
@@ -1597,7 +1599,8 @@ private void calculateInputs(BuildStep step) throws CoreException {
 										if ((var = var.trim()).length() != 0) {
 											if (VAR_USER_OBJS.equals(var)) {
 												String objs[] = getUserObjs(step);
-												if (objs != null && objs.length != 0) {
+												// if there are user objects and they will not be provided as extra flags
+												if ((!toolHasExtraFlags) && (objs != null) && (objs.length != 0)) {
```

Comment:
Have we started using a new code style that includes redundant parenthesis?
In my book, this should be:
```
if (!toolHasExtraFlags && objs != null && objs.length != 0) {
    ...
}
```
But I might have missed something...

Reply:
We must have different books. :-)